src/lua/digest.lua

```diff
@@ -267,4 +313,70 @@ m['aes256cbc'] = {
     end
 }
 
+for _, var in ipairs({'32', '64'}) do
+    local xxHash
+
+    local update_fn_name = 'XXH' .. var .. '_update'
+    local digest_fn_name = 'XXH' .. var .. '_digest'
+    local reset_fn_name = 'XXH' .. var .. '_reset'
+    local copy_fn_name = 'XXH' .. var .. '_copyState'
+    local methods = {
+        default_seed = 0,
+
+        update = function(self, str)
+            if type(str) ~= 'string' then
+                error("Usage xxhash" .. var .. ":update(string)")
+            end
+            return builtin[update_fn_name](self, str, #str) == builtin.XXH_OK
+        end,
+        result = function(self)
+            return builtin[digest_fn_name](self)
+        end,
+        clear = function(self, seed)
+            if seed == nil then
+                seed = self.default_seed
+            end
+            return builtin[reset_fn_name](self, seed) == builtin.XXH_OK
+        end,
+        copy = function(self)
+            local copy = xxHash.new()
+            builtin[copy_fn_name](copy, self)
+            return copy
+        end,
+    }
+
+    local XXH_state_t = ffi.typeof('XXH' .. var .. '_state_t')
+    ffi.metatype(XXH_state_t, {
+        __index = function(_, key)
+            return methods[key]
+        end,
+        __tostring = function(self)
```

Comment:
Do you really need `__tostring()`? I don't see it used in other hashes.

Reply:
It used to show readable value in console. Removed.